homeassistant/components/nam/sensor.py

```diff
@@ -71,238 +73,291 @@
     ),
     NAMSensorEntityDescription(
         key=ATTR_SPS30_P0,
         name="SPS30 particulate matter 1.0",
+        native_precision=0,
         native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
         device_class=SensorDeviceClass.PM1,
         state_class=SensorStateClass.MEASUREMENT,
+        value=lambda sensors: sensors.sps30_p0,
     ),
     NAMSensorEntityDescription(
         key=ATTR_SPS30_P1,
         name="SPS30 particulate matter 10",
+        native_precision=0,
         native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
         device_class=SensorDeviceClass.PM10,
         state_class=SensorStateClass.MEASUREMENT,
+        value=lambda sensors: sensors.sps30_p1,
     ),
     NAMSensorEntityDescription(
         key=ATTR_SPS30_P2,
         name="SPS30 particulate matter 2.5",
+        native_precision=0,
         native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
         device_class=SensorDeviceClass.PM25,
         state_class=SensorStateClass.MEASUREMENT,
+        value=lambda sensors: sensors.sps30_p2,
     ),
     NAMSensorEntityDescription(
         key=ATTR_SPS30_P4,
         name="SPS30 particulate matter 4.0",
+        native_precision=0,
         native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
         icon="mdi:molecule",
         state_class=SensorStateClass.MEASUREMENT,
+        value=lambda sensors: sensors.sps30_p4,
     ),
     NAMSensorEntityDescription(
         key=ATTR_DHT22_HUMIDITY,
         name="DHT22 humidity",
+        native_precision=1,
         native_unit_of_measurement=PERCENTAGE,
         device_class=SensorDeviceClass.HUMIDITY,
         state_class=SensorStateClass.MEASUREMENT,
+        value=lambda sensors: sensors.dht22_humidity,
     ),
     NAMSensorEntityDescription(
         key=ATTR_DHT22_TEMPERATURE,
         name="DHT22 temperature",
+        native_precision=1,
         native_unit_of_measurement=UnitOfTemperature.CELSIUS,
         device_class=SensorDeviceClass.TEMPERATURE,
         state_class=SensorStateClass.MEASUREMENT,
+        value=lambda sensors: sensors.dht22_temperature,
     ),
     NAMSensorEntityDescription(
         key=ATTR_SIGNAL_STRENGTH,
         name="Signal strength",
+        native_precision=0,
         native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
         device_class=SensorDeviceClass.SIGNAL_STRENGTH,
         entity_registry_enabled_default=False,
         state_class=SensorStateClass.MEASUREMENT,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda sensors: sensors.signal,
     ),
     NAMSensorEntityDescription(
         key=ATTR_UPTIME,
         name="Uptime",
         device_class=SensorDeviceClass.TIMESTAMP,
         entity_registry_enabled_default=False,
         entity_category=EntityCategory.DIAGNOSTIC,
+        value=lambda sensors: utcnow()
+        - timedelta(seconds=getattr(sensors, "uptime", 0)),
```

Comment:
Multi-line lambdas should be avoided. Does this work?
```suggestion
        value=lambda sensors: utcnow() - timedelta(seconds=sensors.uptime or 0),
```

Reply:
Works, thanks.